### PR TITLE
Upgrade to PHP 8.1

### PR DESCRIPTION
**Pull request by zepfietje**
Waiting for https://github.com/filamentphp/filament/pull/6453 to be merged first.

### CODE REVIEW

File: packages/app/src/Commands/MakePageCommand.php [hunk 1]
```diff
@@ -115,10 +115,10 @@ public function handle(): int
                 ->append('.blade.php'),
         );
 
-        $files = array_merge(
-            [$path],
-            $resourcePage === 'custom' ? [$viewPath] : [],
-        );
+        $files = [
+            $path,
+            ...$resourcePage === 'custom' ? [$viewPath] : [],
```
Comment (pxlrbt) [inline packages/app/src/Commands/MakePageCommand.php]: I think for better readability we should wrap this in parentheses when used with ternary operator:

```suggestion
            ...($resourcePage === 'custom' ? [$viewPath] : []),
```

File: packages/app/src/Pages/Concerns/HasRoutes.php [hunk 2]
```diff
@@ -35,11 +35,11 @@ public static function getSlug(): string
...
     public static function getRouteMiddleware(Context $context): string | array
     {
-        return array_merge(
-            (static::isEmailVerificationRequired($context) ? [static::getEmailVerifiedMiddleware($context)] : []),
-            (static::isTenantSubscriptionRequired($context) ? [static::getTenantSubscribedMiddleware($context)] : []),
-            static::$routeMiddleware,
-        );
+        return [
+            ...static::isEmailVerificationRequired($context) ? [static::getEmailVerifiedMiddleware($context)] : [],
+            ...static::isTenantSubscriptionRequired($context) ? [static::getTenantSubscribedMiddleware($context)] : [],
```
Comment (pxlrbt) [inline packages/app/src/Pages/Concerns/HasRoutes.php]:
```suggestion
            ...(static::isEmailVerificationRequired($context) ? [static::getEmailVerifiedMiddleware($context)] : []),
            ...(static::isTenantSubscriptionRequired($context) ? [static::getTenantSubscribedMiddleware($context)] : []),
```

File: packages/app/src/Resources/Pages/CreateRecord.php [hunk 3]
```diff
@@ -180,11 +180,11 @@ protected function mutateFormDataBeforeCreate(array $data): array
...
     protected function getFormActions(): array
     {
-        return array_merge(
-            [$this->getCreateFormAction()],
-            static::canCreateAnother() ? [$this->getCreateAnotherFormAction()] : [],
-            [$this->getCancelFormAction()],
-        );
+        return [
+            $this->getCreateFormAction(),
+            ...static::canCreateAnother() ? [$this->getCreateAnotherFormAction()] : [],
```
Comment (pxlrbt) [inline packages/app/src/Resources/Pages/CreateRecord.php]:
```suggestion
            ...(static::canCreateAnother() ? [$this->getCreateAnotherFormAction()] : []),
```

File: packages/app/src/Resources/Pages/Page.php [hunk 4]
```diff
@@ -57,10 +57,10 @@ public function getBreadcrumbs(): array
 
         $breadcrumb = $this->getBreadcrumb();
 
-        return array_merge(
-            [$resource::getUrl() => $resource::getBreadcrumb()],
-            (filled($breadcrumb) ? [$breadcrumb] : []),
-        );
+        return [
+            $resource::getUrl() => $resource::getBreadcrumb(),
+            ...filled($breadcrumb) ? [$breadcrumb] : [],
```
Comment (pxlrbt) [inline packages/app/src/Resources/Pages/Page.php]:
```suggestion
            ...(filled($breadcrumb) ? [$breadcrumb] : []),
```

File: packages/support/src/Components/ViewComponent.php [hunk 5]
```diff
@@ -162,13 +165,13 @@ public function render(): View
...
-                $this->extractPublicProperties(),
-                $this->extractPublicMethods(),
-                isset($this->viewIdentifier) ? [$this->viewIdentifier => $this] : [],
-                $this->viewData,
-            ),
+            [
+                'attributes' => new ComponentAttributeBag(),
+                ...$this->extractPublicProperties(),
+                ...$this->extractPublicMethods(),
+                ...isset($this->viewIdentifier) ? [$this->viewIdentifier => $this] : [],
```
Comment (pxlrbt) [inline packages/support/src/Components/ViewComponent.php]:
```suggestion
                ...(isset($this->viewIdentifier) ? [$this->viewIdentifier => $this] : []),
```

**Comment by zepfietje** [comment]
Hmm I'm not too sure about wrapping, @pxlrbt.
It's already readable to me.

Will let @danharrin decide.

Edit: would need to be added for null coalescing too then.

**Comment by danharrin** [comment]
I like the parentheses to clarify what expression is being spread

**Pull request merged**